src/Parser/Druid/Feral/Modules/FeralCore/Snapshot.js

```diff
@@ -0,0 +1,127 @@
+const PANDEMIC_FRACTION = 0.3;
+
+// leeway in ms after loss of bloodtalons/prowl buff to count a cast as being buffed. Keep well below GCD to avoid false positives.
+const BUFF_WINDOW_TIME = 300;
+
+class Snapshot extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  // extending class should fill these in:
+  spellCastId;
+  isProwlAffected;
+  isTigersFuryAffected;
+  isBloodtalonsAffected;
+  durationOfFresh;
+
+  // initialised to before combat log starts to avoid false positives
+  bloodtalonsFadedAt = -10000;
+  prowlFadedAt = -10000;
+
+  stateByTarget = [];
+
+  // It's common for buffs that are consumed by an ability to be reported as not present when the ability's event is processed.
+  // So record when the buff wears off, and if that's sufficiently close to the ability being used we can assume it was active.
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (this.isProwlAffected && (
+        SPELLS.PROWL.id === spellId ||
+        SPELLS.PROWL_INCARNATION.id === spellId || 
+        SPELLS.SHADOWMELD.id === spellId)) {
+      this.prowlFadedAt = event.timestamp;
+    } else if (this.isBloodtalonsAffected && SPELLS.BLOODTALONS_BUFF.id === spellId) {
+      this.bloodtalonsFadedAt = event.timestamp;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    if (this.spellCastId !== event.ability.guid) {
+      return;
+    }
+    this.dotApplied(event);
+  }
+
+  dotApplied(event) {
+    const targetString = encodeTargetString(event.targetID, event.targetInstance);
+    const stateOld = this.stateByTarget[targetString];
+    const timeRemainOnOld = stateOld ? (stateOld.expireTime - event.timestamp) : 0;
+    const stateNew = this.makeNewState(event, timeRemainOnOld);
+    this.stateByTarget[targetString] = stateNew;
+    debug && console.log(`DoT ${this.spellCastId} applied at ${this.owner.formatTimestamp(event.timestamp)} Prowl:${stateNew.prowl}, TF: ${stateNew.tigersFury}, BT: ${stateNew.bloodtalons}. Expires at ${this.owner.formatTimestamp(stateNew.expireTime)}, pandemic from: ${this.owner.formatTimestamp(stateNew.pandemicTime)}`);
+
+    if (timeRemainOnOld > 0) {
+      this.checkRefreshRule(event, stateOld, stateNew);
+    }
+  }
+
+  makeNewState(event, timeRemainOnOld) {
+    const combatant = this.combatants.selected;
+
+    let expireNew = event.timestamp + this.durationOfFresh;
+    if (timeRemainOnOld > 0) {
+      expireNew += Math.min(this.durationOfFresh * PANDEMIC_FRACTION, timeRemainOnOld);
+    }
+
+    const stateNew = {
+      expireTime: expireNew,
+      pandemicTime: expireNew - this.durationOfFresh * PANDEMIC_FRACTION, // after this time the DoT is in the "pandemic window"
+      tigersFury: false,
+      prowl: false,
+      bloodtalons: false,
+      power: 1,
+    };
+    if (this.isProwlAffected && (combatant.hasBuff(SPELLS.INCARNATION_KING_OF_THE_JUNGLE_TALENT.id, event.timestamp) ||
```

Comment:
You can use `combatant.hasBuff(SPELLS.INCARNATION_KING_OF_THE_JUNGLE_TALENT.id, null, BUFF_WINDOW_TIME)` instead of saving the timestamps for prowl and blood talons. You can read more about .hasBuff here: https://github.com/WoWAnalyzer/WoWAnalyzer/blob/eb8fcfd489ac1bd7c0a695c6657cfe07a0c27885/src/Parser/Core/Entity.js#L46